internal/pkg/coordinator/monitor.go

```diff
@@ -192,7 +195,9 @@ func (m *monitorT) handlePolicies(ctx context.Context, hits []es.HitT) error {
 func (m *monitorT) ensureLeadership(ctx context.Context) error {
 	m.log.Debug().Msg("ensuring leadership of policies")
 	err := dl.EnsureServer(ctx, m.bulker, m.version, m.agentMetadata, m.hostMetadata, dl.WithIndexName(m.serversIndex))
+
 	if err != nil {
+		m.log.Debug().Err(err).Str("eshost", m.hostMetadata.Name).Msg("Failed to ")
 		return err
```

Comment:
If this error happens, an error on the info level will be logged anyways. Why not directly return the error with the info from the debug log message inside so that also on the info level we get the elasticsearch host information. With this you can remove the Debug log level.

Reply:
i figured additional information could be placed at specific error points with more information, returning a formatted error would achieve the same thing though. i'll update the pr.